### ISSUE TITLE
Shouldn't the PDFs of `UnivariateGMM` (and possibly other mixtures) be "squashed" according to their weights?

### ISSUE THREAD

**Issue by ForceBru** [JuliaPlots/Plots.jl#5410]
TL;DR: currently PDFs of the components seem to be plotted without considering their weights (that [the code](https://github.com/JuliaStats/Distributions.jl/blob/8c7f400bd8b9ffe3773f2ee52501ee75b0fba3cd/src/mixtures/unigmm.jl#L7) calls "prior"). Maybe it would be better to plot them weighted, or maybe plot the PDF of the _mixture_ (instead of plotting PDFs of its components) with an option to add PDFs of the _weighted_ components to the plot?

-------

The pull request https://github.com/JuliaPlots/StatsPlots.jl/pull/456 introduced plotting of `UnivariateGMM`, so I was trying to use it to see how well my model describes the data. However, the fit looked terrible (the PDFs of the components were way too spiky) despite convergence criteria and the log-likelihood showing that the fit wasn't that bad. I plotted the data from the pull request's sample code and found that the PDFs of the components are not weighted:

```julia
using StatsPlots
using Distributions

distr = UnivariateGMM(
	[0, 4], [1, 2], Categorical([0.2, 0.8])
)

data = rand(distr, 10_000)

plt = histogram(data, bins=300, normalize=true, linewidth=0)
plot!(plt, distr)
savefig(plt, "stats.png")
```

![stats](https://user-images.githubusercontent.com/8695383/125529510-9b482cbf-4e0c-4f69-87b5-fd670fbb083e.png)

I think the PDF on the left doesn't look anything like the leftmost "bump" of the histogram, so it looks like the mixture model fits the data poorly, yet we know that the data were literally sampled from this exact mixture. I would expect the plots of the two PDFs to sort of "hug" the histogram, like this:

```julia
x = range(minimum(data), maximum(data), length=500)
μs, σs, p_distr = params(distr)
plt2 = histogram(data, bins=300, normalize=true, linewidth=0)
for (μ, σ, p) ∈ zip(μs, σs, probs(p_distr))
	# Weight the PDF by multiplying by `p`
	the_pdf = @. p * exp(-1/2 * (x - μ)^2/σ^2) / (sqrt(2π) * σ)
	plot!(plt2, x, the_pdf, label="μ = $μ", linewidth=3)
end
savefig(plt2, "stats2.png")
```

![stats2](https://user-images.githubusercontent.com/8695383/125530454-8aa91a26-9401-46a0-97ce-35b68930f5ef.png)

Of course, there's overlap in the middle that _individual_ components can't explain that well, but the full PDF of the mixture can:

```julia
x = range(minimum(data), maximum(data), length=500)
full_pdf = zeros(size(x))
μs, σs, p_distr = params(distr)
for (μ, σ, p) ∈ zip(μs, σs, probs(p_distr))
	@. full_pdf += p * exp(-1/2 * (x - μ)^2/σ^2) / (sqrt(2π) * σ)
end
plt3 = histogram(data, bins=300, normalize=true, linewidth=0)
plot!(plt3, x, full_pdf, label="Mixture PDF", linewidth=3)
savefig(plt3, "stats3.png")
```

![stats3](https://user-images.githubusercontent.com/8695383/125530849-6c3c2794-9943-4b83-a8c2-20aa939bf810.png)

--------

So, maybe plot the PDF of the mixture and, optionally, the weighted components' PDFs?
